src/internet_identity/tests/integration/v2_api/authn_method_metadata.rs

```diff
@@ -0,0 +1,201 @@
+use crate::v2_api::authn_method_test_helpers::{
+    create_identity_with_authn_method, test_authn_method,
+};
+use candid::Principal;
+use canister_tests::api::internet_identity::api_v2;
+use canister_tests::framework::{
+    env, expect_user_error_with_message, install_ii_canister, II_WASM,
+};
+use ic_test_state_machine_client::CallError;
+use ic_test_state_machine_client::ErrorCode::CanisterCalledTrap;
+use internet_identity_interface::internet_identity::types::{
+    AuthnMethodMetadataReplaceError, MetadataEntryV2,
+};
+use regex::Regex;
+use serde_bytes::ByteBuf;
+use std::collections::HashMap;
+
+#[test]
+fn should_write_authn_method_metadata() -> Result<(), CallError> {
```

Comment:
maybe add a variant when there is some metadata initially, and the call to `authn_method_metadata_replace` really replaces it?